### ISSUE TITLE
[2.0.5][Balance][Multiplayer] When reviving players should start with full HP/MP

### ISSUE THREAD

**Issue by crkellen** [TurningWheel/Barony#124]
**ISSUE:**
When a player dies, but the remaining part makes it to the next floor, they revive without their max health or mana. Loss of all of their gear, plus experience, causes this to contribute heavily to the snowball effect of constant player death.

**EXPECTED RESULT:**
When a player is revived, they should start with their full HP/MP.

**ADDITIONAL INFO:**
Usually what happens is whoever survives gets all of the XP and becomes a carry, while the rest of the players continue to die repeatedly without much hope to recover.

**Comment by danielnelson** [comment]
I'm very sad to see the pull request was closed.  While death ending a run is fine in single player, I feel that multiplayer could be much improved if there was a way to make the game less punishing.  As it is, I just abandon the game and restart on first player death.

Perhaps we could have some options regarding player death penalties?  Personally, I would prefer if it was possible for players to keep their inventories.

**Comment by crkellen** [comment]
The snowballing effect of a single player's death causes them to be forever behind in experience, loot, and fun.
I too believe it to a critical flaw in the game's design, when playing multiplayer. I believe a community vote, or more discussion would be a good step in the right direction.
null

**Comment by SheridanR** [comment]
Loot and experience are both easily shared and there is no loss of experience, attributes, or skill points on death. Given that this is a roguelike, perhaps the penalty for dying should actually be greater than it is. HP is easily recovered (especially at later stages) so ultimately it's only a minor deterrent.

**Comment by danielnelson** [comment]
There is always the opportunity costs of being dead, which are rather steep in my experience, but the biggest issue for me is that you usually cannot carry all of the items of the dead player, and even if you can it's time consuming which breaks the flow of the game and it's a problem to dispense the items on the next level before being mugged.

> Given that this is a roguelike, perhaps the penalty for dying should actually be greater than it is.

I guess it could be game over for them, but while very roguelike that doesn't make for a great multiplayer experience.  No one wants to sit around without playing for too long.

I think it is something that is player and group specific, everyone I have played with loses interest in replaying the first 3 levels over and over, and wants to move on to more forgiving games.  These are players who have no problem dealing with other fairly difficult games such as vagante.  Maybe those of us that are not as good can have the option of a easier experience?

**Comment by SheridanR** [comment]
I play games too, and I know being dead and waiting to respawn in multiplayer is never fun. Being kicked out of a game for dying feels even worse. That is why I consciously chose to add revive from the beginning. It wasn't even a question in my mind that the game needed it.

The fact remains that the game is a rogue-like and the punishment for death should be harsh. In a single-player game you don't experience a lot of downtime for dying, but you still lose all of your progress, which is also intended.

I'm glad you mentioned players having to carry the equipment of another player when they die, which slows them down, and wastes a lot of time giving it back to the other players. This was known in advance and is part of the intended punishment. It is worth noting that you don't HAVE to rescue a player's entire inventory, or any part of it at all really. Usually it is quite possible to scavenge what you need from any level.

**Comment by WALLOFJUSTICE** [comment]
Agreed with the option to enable more forgiving death. I propose this be an optional server flag that keeps equipped items on the dead player, but drops the rest on the ground, and still reviving with half hp/mp. While I won't begin implementing this option due to the current update works, I will happily review and test any pull requests as they come in.

Akin to the minotaur disable flag, it's not the intended way to play, though enables a more co-op friendly option to ease the learning curve.

**Comment by crkellen** [comment]
I like this idea if the new flag. I feel it strikes a good balance. I like the difficulty if the game, but sometimes you just want that power fantasy.
null
